### PR TITLE
The solution of the 'clickable' problem

### PR DESCRIPTION
**Pull request by koryun23**

### CODE REVIEW

File: src/main/com/aca/homework/week7/clickable/AbstractClickable.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.aca.homework.week7.clickable;
+
+public abstract class AbstractClickable implements Clickable{
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week7/clickable/AbstractClickable.java:3]: the txt file is missing

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

File: src/main/com/aca/homework/week7/clickable/AbstractClickable.java (hunk 1 above)
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week7/clickable/AbstractClickable.java:3]: please format the code (ctrl +alt+shift+L)

File: src/main/com/aca/homework/week7/clickable/AbstractClickable.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    Action action;
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week7/clickable/AbstractClickable.java:4]: should be private

File: src/main/com/aca/homework/week7/clickable/Button.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package com.aca.homework.week7.clickable;
+
+public class Button extends AbstractClickable implements Clickable {
+    String name;
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week7/clickable/Button.java:4]: should be private, also you can move the name to Abstract class

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

File: src/main/com/aca/homework/week7/clickable/AbstractClickable.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        this.action = action;
+        this.name = name;
+    }
+
+    @Override
+    public void click() {
+        action.doAction(this);
+    }
+
+    public String getName() {
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week7/clickable/AbstractClickable.java:17]: just implement the name() method and do not provide getName()

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

**Comment by koryun23** [comment]
done

File: src/main/com/aca/homework/week7/clickable/Button.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package com.aca.homework.week7.clickable;
+
+public class Button extends AbstractClickable implements Clickable {
+    public Button(String name, Action action) {
+        super(name, action);
+    }
+
+    @Override
+    public String name() {
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week7/clickable/Button.java:9]: please remove the name method from this class

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

**Comment by koryun23** [comment]
done

**Pull request merged**